### PR TITLE
Mldbfb 411 clearer expression val error msg

### PR DESCRIPTION
**Pull request by mailletf**

### CODE REVIEW

File: sql/expression_value.h [hunk 1]
```diff
@@ -807,6 +807,31 @@ struct ExpressionValue {
 
     Type type_;
 
+    static inline std::string print(Type t) {
+        switch (t)  {
+        case NONE:      return "empty";
+        case ATOM:      return "atomic value";
```
Comment (guyd) [inline sql/expression_value.h]: Although _atom_ is still present in the documentation, I thought we were using _scalar_ now.  @nicolaskruchten ?

File: sql/expression_value.h [hunk 2]
```diff
@@ -807,6 +807,31 @@ struct ExpressionValue {
...
+        case ROW:       return "row";
+        case STRUCT:    return "struct";
+        case EMBEDDING: return "embedding";
+        default:
+            throw ML::Exception("Unknown ExpressionValue type: " + t);
```
Comment (guyd) [inline sql/expression_value.h]: If this is targeted to user, I would not mention the name of the class here.

File: sql/expression_value.h [hunk 3]
```diff
@@ -807,6 +807,31 @@ struct ExpressionValue {
...
+        }
+    }
+
+    void inline assertType(Type requested, const std::string & details="") const {
+        if(requested != type_) {
+            std::string msg = "Cannot convert ExpressionValue of type "
```
Comment (guyd) [inline sql/expression_value.h]: and here

**Comment by guyd** [comment]
+1 with minor comments

File: sql/expression_value.h (hunk 1 above)
Comment (nicolaskruchten) [inline sql/expression_value.h]: atomic value is ok for now, we still haven't fully standardized :)

**Pull request merged**